src/components/FlagsProvider.tsx

```diff
@@ -31,9 +32,21 @@ const FlagsProvider = ({children, config}: FlagsProviderProps) => {
         const flagsClientInstance = new FlagsClient({...defaultConfig, ...config });
         await flagsClientInstance.init();
 
+        if (!isMountedRef.current) {
```

Comment:
If you move this to the top of the function, before the FlagsClient obj is created, is doen't have to create and initiate the FlagsClient when it is not mounted.